web/beacon-app/src/features/topics/components/QueryForm.tsx

```diff
@@ -1,36 +1,49 @@
 import { Trans } from '@lingui/macro';
 import { Button, TextField } from '@rotational/beacon-core';
+import { Form, FormikHelpers, FormikProvider } from 'formik';
+
+import { ProjectQueryDTO } from '@/features/projects/types/projectQueryService';
+
+import { useTopicQueryInputForm } from '../schemas/topicQueryInputValidationSchema';
 
 type QueryFormProps = {
   defaultEnSQL: string;
-  queryHandler: (query: string) => void;
+  isSubmitting?: boolean;
+  onSubmit: (values: ProjectQueryDTO, formikHelpers: FormikHelpers<ProjectQueryDTO>) => void;
 };
 
-const QueryForm = ({ defaultEnSQL, queryHandler }: QueryFormProps) => {
-  const onQuery = (e: any) => {
-    e.preventDefault();
-    queryHandler(e.target.value);
-  };
+const QueryForm = ({ defaultEnSQL, isSubmitting, onSubmit }: QueryFormProps) => {
+  const formik = useTopicQueryInputForm(onSubmit);
+
+  const { handleReset, handleSubmit, getFieldProps, touched, errors } = formik;
 
   return (
-    <div className="mt-4 flex space-x-2">
-      <TextField
-        labelClassName="font-semibold"
-        type="search"
-        defaultValue={defaultEnSQL}
-        onChange={onQuery}
```

Comment:
`defaultValue` doesn't display anything, possibly because the initial value in the form options is `' '`, so it was removed. `onChange` caused the query request to fire immediately after a user makes a change, so this is why it was removed.